modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/upgrade/v0_1_0/test/DeleteAPIPropertiesToAPIPropertiesUpgradeProcessTest.java

```diff
@@ -0,0 +1,130 @@
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.test.rule.PermissionCheckerMethodTestRule;
+import com.liferay.portal.tools.DBUpgrader;
+import com.liferay.portal.upgrade.registry.UpgradeStepRegistrator;
+import com.liferay.portal.upgrade.test.util.UpgradeTestUtil;
+import com.liferay.portal.vulcan.util.LocalizedMapUtil;
+
+import org.apache.commons.lang.time.StopWatch;
+
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+import org.springframework.util.Assert;
+
+/**
+ * @author Alejandro Tardín
+ */
+@FeatureFlags("LPS-178642")
+public class DeleteAPIPropertiesToAPIPropertiesUpgradeProcessTest
+	extends BaseTestCase {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@BeforeClass
+	public static void setUpClass() {
+		_originalStopWatch = ReflectionTestUtil.getFieldValue(
+			DBUpgrader.class, "_stopWatch");
+	}
+
+	@AfterClass
+	public static void tearDownClass() {
+		ReflectionTestUtil.setFieldValue(
+			DBUpgrader.class, "_stopWatch", _originalStopWatch);
+	}
+
+	@Before
+	public void setUp() throws Exception {
+		super.setUp();
+
+		ReflectionTestUtil.setFieldValue(DBUpgrader.class, "_stopWatch", null);
+	}
+
+	@Test
+	public void testUpgradeProcess() throws Exception {
+		ObjectDefinition objectDefinition =
+			_objectDefinitionLocalService.
+				fetchObjectDefinitionByExternalReferenceCode(
+					"L_API_PROPERTY", TestPropsValues.getCompanyId());
+
+		_objectRelationshipLocalService.addObjectRelationship(
+			"L_API_PROPERTIES_TO_API_PROPERTIES", TestPropsValues.getUserId(),
+			objectDefinition.getObjectDefinitionId(),
+			objectDefinition.getObjectDefinitionId(), 0,
+			ObjectRelationshipConstants.DELETION_TYPE_PREVENT,
+			LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+			StringUtil.randomId(), true,
+			ObjectRelationshipConstants.TYPE_MANY_TO_MANY, null);
+
+		UpgradeProcess upgradeProcess = UpgradeTestUtil.getUpgradeStep(
+			_upgradeStepRegistrator,
+			"com.liferay.headless.builder.internal.upgrade.v0_1_0." +
+				"DeleteAPIPropertiesToAPIPropertiesUpgradeProcess");
+
+		String liferayMode = SystemProperties.get("liferay.mode");
+
+		try {
+			SystemProperties.clear("liferay.mode");
+
+			StartupHelperUtil.setUpgrading(true);
+
+			upgradeProcess.upgrade();
+		}
+		finally {
+			SystemProperties.set("liferay.mode", liferayMode);
+
+			StartupHelperUtil.setUpgrading(false);
+		}
+
+		Assert.isNull(
+			_objectRelationshipLocalService.
+				fetchObjectRelationshipByExternalReferenceCode(
+					"L_API_PROPERTIES_TO_API_PROPERTIES",
+					objectDefinition.getObjectDefinitionId()));
+	}
+
+	private static StopWatch _originalStopWatch;
```

Comment:
I would rename it to `_stopWatch` as there are no more stopWatch

Reply:
I based it on https://github.com/liferay/liferay-portal/blob/dd44186cbc815ece23a21f8402470fbaafebeb33/modules/apps/portal/portal-upgrade-test/src/testIntegration/java/com/liferay/portal/upgrade/test/UpgradeRecorderTest.java#L452.